server/docker-compose.yml

```diff
@@ -0,0 +1,48 @@
+version: "3.8"
+services:
+  hwapi-dev:
+    build:
+      context: .
+      dockerfile: Dockerfile
+    command: ./scripts/dev-docker-entrypoint.sh
+    working_dir: /home/app
+    volumes:
+      - .:/home/app
+      - ./data:/home/app/data
+    ports:
+      - "8080:8080"
+    environment:
+      - DB_URL=sqlite:////home/app/data/hwapi.db
+
+  hwapi-test:
+    build:
+      context: .
+      dockerfile: Dockerfile
+    command: >
+      bash -c '
+        python scripts/seed_db.py
+        black --check hwapi/ scripts/ tests/
+        ruff check hwapi/ scripts/ tests/
+        mypy hwapi/ scripts/ tests/
+        pytest'
+    working_dir: /home/app
+    volumes:
+      - .:/home/app
+      - ./data:/home/app/data
```

Comment:
Since this is for the hwapi-test, then you probably don't want to reuse data that could be there from a dev environment or even a previous test environment. And you also don't need to leave the data around. What do you think about creating a tmpfs volume so that you get a clean copy of the test data each time, and there's no chance it could interfere?
You could do that with:
```
    tmpfs:
      - /home/app/data
```